packages/playwright-core/src/server/frames.ts

```diff
@@ -1279,6 +1279,8 @@ export class Frame extends SdkObject {
       // A: We want user to receive a friendly message containing the last intermediate result.
       if (js.isJavaScriptErrorInEvaluate(e))
         throw e;
+      if (!isSessionClosedError(e))
```

Comment:
This looks suspicious, what's the exact scenario?

Reply:
Selector with syntax error or non-existent engine.